File: Data Structures/js-data-structures/queues.js
// Queue

class Queue{
    constructor(){
        this.items = {};
        this.frontIndex = 0;
        this.backIndex = 0;
    }

    enqueue(item){      // add elements at the end of queue
        this.items[this.backIndex] = item
        this.backIndex++
        return item 
    }

    dequeue(){      // remove elements at the front of queue
        const item = this.items[this.frontIndex]
        delete this.items[this.frontIndex]
        this.frontIndex++
        return item
    }

    peek(){     // peek the front element of the queue
        return `front -> ${this.items[this.frontIndex]}`
    }

    size(){     // return the size of queue
        return this.backIndex - this.frontIndex
    }

    printQueue(){       // print queue
        return this.items
    }

}

const queue = new Queue();

console.log(queue.enqueue(10));
console.log(queue.enqueue(20));
console.log(queue.enqueue(30));
queue.dequeue()

console.log(queue.peek())
console.log(queue.size())

const print = queue.printQueue()
console.log(print)

File: Data Structures/py-data-structures/queues.py
# Queue

class Queue:
    def __init__(self):
        self.items = {}
        self.front_index = 0
        self.back_index = 0
        
    def enqueue(self, item):    # add elements at the end of queue
        self.items[self.back_index] = item
        self.back_index += 1
        return item
    
    def dequeue(self):      # remove elements at the front of queue
        item = self.items[self.front_index]
        del self.items[self.front_index]
        self.front_index += 1
        return item

    def peek(self):     # peek the front element of the queue
        return f"front -> {self.items[self.front_index]}"
    
    def size(self):     # return the size of queue
        return self.back_index - self.front_index
    
    def print_queue(self):      # print queue
        return self.items
    

queue = Queue()

print(queue.enqueue(10))
print(queue.enqueue(20))
print(queue.enqueue(30))
queue.dequeue()
print(queue.peek())
print(queue.size())
print(queue.print_queue())